src/components/input/README.md

```diff
@@ -38,5 +46,76 @@ You specify a hint-label in one of two ways; either using the `hintLabel` attrib
 
 Specifying a side twice will result in an exception during initialization.
 
+#### Example
+
+A simple character counter can be made like the following:
+
+```html
+<md-input placeholder="Character count (100 max)" maxLength="100" class="demo-full-width"
+          value="Hello world. How are you?"
+          #characterCountHintExample>
+  <md-hint align="end">{{characterCountHintExample.characterCount}} / 100</md-hint>
+</md-input>
+```
+
+<img src="https://material.angularjs.org/material2_assets/input/character-count.png">
+
+
+
 ## Divider Color
 
+The divider (line under the <md-input> content) color can be changed by using the `dividerColor` attribute. A value of `primary` is the default and will correspond to your theme primary color. Alternatively, you can specify `accent` or `warm`.
+
+#### Example
+
+^((please not that this example has been doctored to show the colors; they would normally show only on focus)^)
+
+<img src="https://material.angularjs.org/material2_assets/input/divider-colors.png">
+
+
+
+## Labelling
+
+You can label the `<md-input>` as you would a regular `<input>`.
+
+
+
+## Full Forms
+
+You can make a full form using inputs, and it will support autofill natively.
+
+#### Example
+
+```html
+<md-card class="demo-card demo-basic">
+  <md-toolbar color="primary">Basic</md-toolbar>
+  <md-card-content>
+    <form>
+      <md-input class="demo-full-width" placeholder="Company (disabled)" disabled value="Google">
+      </md-input>
+
+      <table style="width: 100%" cellspacing="0"><tr>
+        <td><md-input placeholder="First name" style="width: 100%"></md-input></td>
+        <td><md-input placeholder="Long Last Name That Will Be Truncated" style="width: 100%"></md-input></td>
+      </tr></table>
+      <p>
+        <md-input class="demo-full-width" placeholder="Address" value="1600 Amphitheatre Pkway"></md-input>
+        <md-input class="demo-full-width" placeholder="Address 2"></md-input>
+      </p>
+      <table style="width: 100%" cellspacing="0"><tr>
+        <td><md-input class="demo-full-width" placeholder="City"></md-input></td>
+        <td><md-input class="demo-full-width" placeholder="State"></md-input></td>
+        <td><md-input #postalCode class="demo-full-width" maxLength="5"
+                      placeholder="Postal Code"
+                      value="94043">
+          <md-hint align="end">{{postalCode.characterCount}} / 5</md-hint>
+        </md-input></td>
+      </tr></table>
+    </form>
+  </md-card-content>
+</md-card>
+```
+
+Will result in this:
+
+<img src="https://material.angularjs.org/material2_assets/input/full-form.png">
```

Comment:
I saw that, but was wondering if I should add padding myself or let the user handle the padding?

The text flow in if we have no padding, in certain cases maybe the user wants that. If so, I can just add a `&nbsp;` in the example.